src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/MemoryClassLoader.java

```diff
@@ -271,6 +267,25 @@ public URL nextElement() {
         };
     }
 
+    /**
+     * Resolves a "resource name" (as used in the getResource* methods)
+     * to an existing file relative to source root path, or null otherwise.
+     *
+     * @param name the resource name
+     * @return the URL of the resource, or null
+     */
+    private URL toResourceInRootPath(String name) {
+        var file = programDescriptor.sourceRootPath().resolve(name);
```

Comment:
"name" is an abstract resource name, so it's a '/' separated sequence of names. Is there is mapping step missing here?

Reply:
Hm, if it missing, it was missing before. I moved the line from the former location with a different variable name:
`var programPath = programDescriptor.sourceRootPath().resolve(name);` into this new helper method. All tests and manual usages remain "green".

Which kind of mapping are you missing here? From `/` to `\\` or `.`?

Reply:
In java.base, look at jdk.internal.module.Resources.toFilePath to see how the built-in class loaders map resource names to relative file paths. I don't think you'll need as much here. My point is that a resource name can't be be used a file path. Also it's okay if you want to separate it to a separate issue, it's just something that jumps out when scanning this.

Reply:
Right. With `jdk.internal.module.Resources` being already used in the `MemoryModuleFinder` sibling, I'll replace the too simple `.resolve(name)` as suggested and in this PR.